parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -455,6 +459,35 @@ data ForkFailure = SrcNotFound | DestExists
 numHashChars :: Branch m -> Int
 numHashChars _b = 3
 
+-- This type is a little ugly, so we wrap it up with a nice type alias for
+-- use outside this module.
+type Cache m = Cache.Cache m (Causal.RawHash Raw) (Causal m Raw (Branch0 m))
+
+boundedCache :: MonadIO m => Word -> m (Cache m)
+boundedCache = Cache.semispaceCache
```

Comment:
Added this type alias and this `boundedCache` helper to reduce API surface area a bit. You create a `Branch.Cache m` using `Branch.boundedCache`, and then pass that to `Branch.cachedRead`. No need to import other modules.